### ISSUE TITLE
[Bug]: F10 key not mapping properly...

### ISSUE THREAD

**Issue by tenko23** [Yimura/YimMenu#3672]
### Describe the bug

When the 'Teleport to Objective' hotkey is set to the F10 (function 10) key, teleporting to the objective does not work... however, the 'Teleport to Objective' button within the menu does work as expected (as well as the hotkey for teleporting to a map pin -- in my case, the key F9).  The objective hotkey was not tested with other keys.

### Steps To Reproduce

1.  Go to 'Settings' within the menu.
2. Click on 'Hotkeys'.
3. Set the 'Teleport to Objective' hotkey to F10.
4. Exit the menu.
5. Start a setup, etc., with a yellow objective marker on the map.
6. Press the F10 ('Teleport to Objective') hotkey.
7. Observe that the 'Teleport to Objective' hotkey does not work; however the objective button within the menu will.

### Expected Behavior

One should be able to teleport to the objective by using the hotkey without issue.

### Logs

[cout.txt](https://github.com/user-attachments/files/16920149/cout.txt)

### Store

Steam

### Language

English

### YimMenu Language

English (US)

### Additional context

The 'Teleport to Objective' hotkey was not tested with any other keys.  This hotkey issue has been an issue for at least a number of months now.

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by xesdoog** [comment]
This happened to me too, it did nothing when I set it to F10 so I changed it to F11

**Comment by tenko23** [comment]
Interesting... I already have a hotkey set for F11, so I'll flip the two around and see what happens :) .

**Comment by tenko23** [comment]
Yeah, it seems that F10 is just dead period - after I swapped the two... so it may be that the menu is not mapping F10 properly.

**Comment by xesdoog** [comment]
> Yeah, it seems that F10 is just dead period - after I swapped the two... so it may be that the menu is not mapping F10 properly.

it's because in Win32 F10 and Left ALT are system keys and can only be detected using `WM_SYSKEYDOWN`

**Comment by tenko23** [comment]
That could be part of the issue.  I remember that I ran into something similar with Kideons... I had to manually enter the F-keys in decimal form within the settings file, in order for them to work (F10 did work there... maybe F11 didn't?... don't remember anymore -- been too long ;p ).  I see that within Yim's setting file, it's in decimal form as well and does get saved, yet cease to work 🤷🏼‍♂️ .

**Comment by xesdoog** [comment]
> That could be part of the issue. I remember that I ran into something similar with Kideons... I had to manually enter the F-keys in decimal form within the settings file, in order for them to work (F10 did work there... maybe F11 didn't?... don't remember anymore -- been too long ;p ). I see that within Yim's setting file, it's in decimal form as well and does get saved, yet cease to work 🤷🏼‍♂️ .

If a YimMenu dev sees this comment they'll probably roast my ass but what I think is happening is when the menu checks for a pressed button to execute the feature, it's not checking for `WM_SYSKEYDOWN` but the menu does check it while setting the hotkey, hence why it can be set but can not be used.
Left ALT behaves the same as F10 and they are the only syskeys afaik

**Comment by tenko23** [comment]
I'm a Mac guy -- only use Windows to play games, so these interesting Windows quirks (some that seem to have lasted for decades) are always fun to know about 'n learn ;) .....
